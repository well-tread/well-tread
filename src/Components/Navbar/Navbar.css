
/* #container{
    
  }
  
  #container:after{
   content: '';
   width: 300px;
   height: 300px;
   background: inherit; 
   position: absolute;
   left: 0px;
   right: 0;
   top: 0px;  
   bottom: 0;
   box-shadow: inset 0 0 0 200px rgba(255,255,255,0.05);
   filter: blur(30px);
  } */
  
  /* #container{
      width: 100%;
      height: 7%;
      filter: blur(2px);
      background-color: rgba(161, 161, 161, 0.582);
      top: 0;
      z-index: 9999;
  }
  #container:after{
      width: 100%;
      height: 7%;
      filter: blur(10px);
      background-color: rgba(131, 131, 131, 0.301);
      position: relative;
      top: 0;
  }
  #container:before{
      width: 100%;
      height: 7%;
      /* filter: blur(10px);
      background-color: rgba(131, 131, 131, 0.301);
      position: relative;
      top: 0;
  } */
