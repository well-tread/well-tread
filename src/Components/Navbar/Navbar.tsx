import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import './Navbar.css';

// Material-UI Core Imports
import { createStyles, Theme, withStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import IconButton from '@material-ui/core/IconButton';
import Typography from '@material-ui/core/Typography';
import SwipeableDrawer from '@material-ui/core/SwipeableDrawer';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import { ButtonProps } from '@material-ui/core/Button';

// Material-UI Icon Imports
import Menu from '@material-ui/icons/Menu';
import Home from '@material-ui/icons/Home';
import AccountBox from '@material-ui/icons/AccountBox';
import Search from '@material-ui/icons/Search';

const styles = (theme: Theme) =>
  createStyles({
    appBarContainer: {
      marginBottom: theme.spacing.unit * 6
    },
    appBar: {
      display: 'flex',
      flexDirection: 'row',
      justifyContent: 'space-between',
      paddingLeft: '2em',
      addingRight: '2em',
      alignItems: 'center',
      // background: '0,0,0,00',
      background: 'white',
      position: 'fixed',
      zIndex: 9999,
      boxShadow: 'none'
    },

    dropDownMenu: {
      paddingTop: '50px',
      [theme.breakpoints.up('sm')]: {
        paddingRight: '80px'
      }
    },
    topNavDropdown: {
      [theme.breakpoints.up('sm')]: {
        display: 'none'
      }
    },
    rightNavDropdown: {
      [theme.breakpoints.down('xs')]: {
        display: 'none'
      },
      [theme.breakpoints.up('sm')]: {
        display: 'block'
      }
    }
  });

export interface Props {
  classes: {
    appBarContainer: string;
    appBar: string;
    dropDownMenu: string;

    topNavDropdown: string;
    rightNavDropdown: string;
  };
}

export interface State {
  drawerOpen: boolean;
  navLinkColor: string;
}

export interface LinkButtonProps extends ButtonProps {
  to: string;
  replace?: boolean;
}

const LinkItem = (props: LinkButtonProps) => (
  <ListItem {...props} component={Link as any} />
);

class Navbar extends Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      drawerOpen: false,
      navLinkColor: 'primary'
    };
    this.toggleDrawer = this.toggleDrawer.bind(this);
  }

  toggleDrawer = (open: boolean) => {
    this.setState({
      drawerOpen: !this.state.drawerOpen
    });
  };

  render() {
    const { classes } = this.props;

    return (
      <div>
        <div className={classes.appBarContainer}>
          <AppBar className={classes.appBar}>
            <Typography component='h6' variant='h6' color='primary'>
              WELL TREAD
            </Typography>
            <IconButton onClick={() => this.toggleDrawer(true)}>
              <Menu color='secondary' className={this.state.navLinkColor} />
            </IconButton>
          </AppBar>
        </div>
        <SwipeableDrawer
          anchor='top'
          className={classes.topNavDropdown}
          open={this.state.drawerOpen}
          onClose={() => this.toggleDrawer(false)}
          onOpen={() => this.toggleDrawer(true)}
        >
          <List component='nav' className={classes.dropDownMenu}>
            <LinkItem to='/home' onClick={() => this.toggleDrawer(false)}>
              <ListItemIcon>
                <Home className={this.state.navLinkColor} />
              </ListItemIcon>
              <ListItemText primary='Home' />
            </LinkItem>
            <LinkItem to='/account' onClick={() => this.toggleDrawer(false)}>
              <ListItemIcon>
                <AccountBox />
              </ListItemIcon>
              <ListItemText primary='Account' />
            </LinkItem>
            <LinkItem to='/search' onClick={() => this.toggleDrawer(false)}>
              <ListItemIcon>
                <Search />
              </ListItemIcon>
              <ListItemText primary='Search for Trails' />
            </LinkItem>
          </List>
        </SwipeableDrawer>
        <SwipeableDrawer
          anchor='right'
          className={classes.rightNavDropdown}
          open={this.state.drawerOpen}
          onClose={() => this.toggleDrawer(false)}
          onOpen={() => this.toggleDrawer(true)}
        >
          <List component='nav' className={classes.dropDownMenu}>
            <LinkItem to='/home' onClick={() => this.toggleDrawer(false)}>
              <ListItemIcon>
                <Home className={this.state.navLinkColor} />
              </ListItemIcon>
              <ListItemText primary='Home' />
            </LinkItem>
            <LinkItem to='/account' onClick={() => this.toggleDrawer(false)}>
              <ListItemIcon>
                <AccountBox />
              </ListItemIcon>
              <ListItemText primary='Account' />
            </LinkItem>
            <LinkItem to='/search' onClick={() => this.toggleDrawer(false)}>
              <ListItemIcon>
                <Search />
              </ListItemIcon>
              <ListItemText primary='Search for Trails' />
            </LinkItem>
          </List>
        </SwipeableDrawer>
      </div>
    );
  }
}

export default withStyles(styles)(Navbar);
