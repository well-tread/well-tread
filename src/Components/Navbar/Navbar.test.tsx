import React from 'react'
import Navbar from './Navbar'
import renderer from 'react-test-renderer'

it('renders the navbar correctly', () =>{
    const tree = renderer
    .create(<Navbar/>)
    .toJSON();
    expect(tree).toMatchSnapshot();
});

