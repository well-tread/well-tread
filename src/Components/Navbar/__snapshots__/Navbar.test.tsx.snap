// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the navbar correctly 1`] = `
Array [
  <div
    className="Navbar-appBarContainer-1"
  >
    <header
      className="MuiPaper-root-16 MuiPaper-elevation4-22 MuiAppBar-root-7 MuiAppBar-positionFixed-8 MuiAppBar-colorPrimary-14 mui-fixed Navbar-appBar-2"
    >
      <h6
        className="MuiTypography-root-43 MuiTypography-h6-60 MuiTypography-colorPrimary-73"
      >
        WELL TREAD
      </h6>
      <button
        className="MuiButtonBase-root-85 MuiIconButton-root-79"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex="0"
        type="button"
      >
        <span
          className="MuiIconButton-label-84"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-88 MuiSvgIcon-colorSecondary-90 primary"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root-111"
        />
      </button>
    </header>
    <div
      className="Navbar-blurBg-3"
    />
  </div>,
  <div
    className="MuiPrivateSwipeArea-root-118 MuiPrivateSwipeArea-anchorTop-121"
    style={
      Object {
        "height": 20,
      }
    }
  />,
  <div
    className="MuiPrivateSwipeArea-root-118 MuiPrivateSwipeArea-anchorRight-120"
    style={
      Object {
        "width": 20,
      }
    }
  />,
]
`;
